Expand docs on state handling with regards to uncommitted non-parameter changes (#9571)
mendix/docs: content/en/docs/refguide/runtime/mendix-client/_index.md

This analysis is done based on the microflow parameters and their usages throughout the microflow. Any time an association is used in the microflow, the association is marked, and will also be sent in the request if needed. In some cases, such as Java actions, the analysis is not done as it would be too performance heavy. In that case, all objects associated with the microflow parameters will be sent.

> When an object is not part of the analyzed state but is retrieved in the microflow within a nonanalyzable context (i.e. in a Java Action or a microflow that was called dynamically in a Java Action) changes made on the client are not applied. This mainly occurs when updating objects associated with the current user or session object in the client that are subsequently retrieved from the database in a Java Action. To prevent issues arising from this, make sure to commit changes to those kinds of objects to the database before calling the microflow.

Reviewer: I initially thought this was in the wrong place, but after reading it a couple of times I think it is in the right place. I think it would help if we made it clearer that we are still talking about microflows so I propose switching the paragraph around.

How about
```suggestion
When an object is retrieved in the microflow using a Java Action or a microflow that was called dynamically in a Java Action, it is not part of the analyzed state and changes made on the client are not applied. This mainly occurs when updating objects in the client associated with the current user or session object that are subsequently retrieved from the database in a Java Action. Make sure to commit changes to those kinds of objects to the database before calling the microflow to prevent issues arising from this.
```

Reviewer: Whoops - missed that this is still awaiting R&D review. Feel free to ignore.

Author: Its a good suggestion, I have applied it.
